### PR TITLE
Compilation optimization for reduce

### PR DESCRIPTION
**Pull request by tianhaodongbd**
### PR types

Others 
### PR changes

Others
### Description

kernel编译体积优化--kernel类型：reduce
Pcard-70459

### CODE REVIEW

**Review by AnnaTrainingG — COMMENTED** [review]
LGTM

**Comment by Xreki** [comment]
更新下最新的develop

File: paddle/phi/kernels/gpu/broadcast_tensors_grad_kernel.cu [hunk 1]
```diff
@@ -88,12 +89,12 @@ void BroadcastTensorsGradKernel(const Context& ctx,
...
-      funcs::ReduceKernel<T, T, kps::AddFunctor, kps::IdentityFunctor<T>>(
-          ctx,
-          *input_tensor,
-          output_tensor,
-          kps::IdentityFunctor<T>(),
-          reduce_dims_vec);
+      phi::SumKernel<T, Context>(ctx,
+                                 *input_tensor,
+                                 reduce_dims_vec,
+                                 input_tensor->dtype(),
```
Comment (Xreki) [inline paddle/phi/kernels/gpu/broadcast_tensors_grad_kernel.cu]: 输出dtype

File: paddle/phi/kernels/legacy/kps/reduce_max_kernel.cu [hunk 2]
```diff
@@ -27,8 +27,7 @@ void MaxRawKernel(const Context& dev_ctx,
                   DenseTensor* out) {
   reduce_all = recompute_reduce_all(x, dims, reduce_all);
   auto out_dtype = x.dtype();
-  phi::Reduce<T, kps::MaxFunctor, kps::IdentityFunctor>(
-      dev_ctx, x, reduce_all, dims.GetData(), keep_dim, out_dtype, out);
+  phi::MaxKernel<T, Context>(dev_ctx, x, dims, keep_dim, out);
```
Comment (Xreki) [inline paddle/phi/kernels/legacy/kps/reduce_max_kernel.cu]: `MaxRawKernel`比`MaxKernel`多一个参数，应在`MaxKernel`里面调`MaxRawKernel`比较合理

File: paddle/phi/kernels/gpu/broadcast_tensors_grad_kernel.cu (hunk 1 above)
Comment (tianhaodongbd) [reply paddle/phi/kernels/gpu/broadcast_tensors_grad_kernel.cu]: done

File: paddle/phi/kernels/legacy/kps/reduce_max_kernel.cu (hunk 2 above)
Comment (tianhaodongbd) [reply paddle/phi/kernels/legacy/kps/reduce_max_kernel.cu]: done

**Review by AnnaTrainingG — APPROVED** [review]

**Pull request merged**